### DISCUSSION TITLE
Restarts and Caliper

### DISCUSSION THREAD

**Discussion by annapiegra** [LLNL/Caliper#384] [Q&A]
Does caliper have any support for restarts: if I need to restart a run is there a way to include the timers and metrics of the original run in the caliper output of the new run?

**Comment by daboehme** [comment]
Hi @annapiegra ,

Unfortunately that's not directly possible right now. However, depending on what output you generate it should be possible to aggregate data from multiple runs afterwards.

**Reply by annapiegra** [reply]
Thanks. I was trying to avoid that route but I'll see if I can create a simple workflow and demo it to the users.

**Reply by gshipman** [reply]
@daboehme If the application carried the prior state of all timers / regions , say in a map or something, is there a straightforward place to capture the first invocation of a particular caliper region, and initialize the caliper values there? perhaps a callback could be added alongside a well defined API for setting a calipers initial value.

**Reply by daboehme** [reply]
Hi @gshipman , sorry about the delay, I meant to get back to this sooner. 

I'd like to understand a bit better what the workflow you have in mind is. Generally, injecting previous profiling output back into Caliper at the start of a run is not entirely trivial, because Caliper typically performs additional data aggregation steps before writing its output. Depending on the specific use case we may be able to come up with something.

However, if the application can carry the entire prior state and also manage the output itself, you could use Caliper's `RegionProfile` class (example here: https://github.com/LLNL/Caliper/blob/master/examples/apps/cali-regionprofile.cpp), which lets you record the region times in an STL map. You could then add the prior values after fetching the current run's times. Would that work for you, or do you need the Caliper-specific output formats for Spot etc.?
